active/0000-ember-data-actions.md

```diff
@@ -0,0 +1,90 @@
+- Start Date: 2015-02-04
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Ember Data wraps typical CRUD actions nicely. However, it is lacking when it comes to endpoints that mutate the state of the world without expecting typical request bodies. We could develop an API that makes interacting with those endpoints easier.
+
+# Motivation
+
+To expose an API to interact with non-CRUD endpoints.
+
+# Detailed design
+
+Imagine that we have a commenting API.
+
+Administrators can curate comments, marking spammy or offensive records as spam.
+
+* `GET /comments` exposes the collection
+* `GET /comments/:id` exposes a particular comment
+* `POST /comments/:id/spam` marks a comment as spam
+
+What is the best way to interact with the `/comments/:id/spam` endpoint?
+
+http://emberjs.jsbin.com/befonozine/1/edit
+
+```javascript
+App.Comment = DS.Model.extend({
+  name: DS.attr("string"),
+  body: DS.attr("string"),
+  
+  reportSpam: function() {
+    var adapter = this.store.adapterFor("comment");
+    var url = adapter.buildURL("comment", this.get("id"), this);
+    
+    return adapter.ajax(url + "/spam", "post");
+  }
+});
+
+App.CommentAdapter = App.ApplicationAdapter.extend({
+  reportSpam: function(comment) {
+    var url = this.buildURL("comment", comment.get("id"), this);
+    
+    return this.ajax(url + "/spam", "post");
```

Comment:
If the server is returning JSON it should return an empty object (`{}`) or empty array depend on what makes more sense. An empty string isn't valid JSON that's why you're seeing the failure. There are arguments in favor and against using 204s to indicate an empty body response.